Makefile

```diff
@@ -338,7 +339,8 @@ DEPS_CFLAGS ?= -MMD -MP -MF .deps/$*.d
 
 %.o: %.c $(CONFIG_H)
 	@mkdir -p .deps/$(*D)
-	$(QUIET_CC)$(CC) -I. -Iinclude $(CFLAGS) $(DEPS_CFLAGS) $(CPPFLAGS) -c -o $@ $<
+	@if [ x$(GENERATE_COMPILATION_DATABASE) = xyes ]; then mkdir -p $(compdb_dir); fi
```

Comment:
Git's Makefile uses

```make
missing_compdb_dir = $(compdb_dir)
$(missing_compdb_dir):
	@mkdir -p $@
```

and then builds objects like

```Makefile
$(C_OBJ): %.o: %.c GIT-CFLAGS $(missing_dep_dirs) $(missing_compdb_dir)
	$(QUIET_CC)$(CC) -o $*.o -c $(dep_args) $(compdb_args) $(ALL_CFLAGS) $(EXTRA_CPPFLAGS) $<
```

however, adding ` $(missing_compdb_dir)` to line 339 didn't work, maybe someone can spot why.